From 93da5ce626ab9e28ee66f9dd230813c769eb9ee4 Mon Sep 17 00:00:00 2001
From: Eyal Reizer <eyalr@ti.com>
Date: Mon, 14 Nov 2011 09:44:10 +0200
Subject: [PATCH 0618/1092] board-am335xevm: changes for working with catalog
 wl1271 com6m module

* The catalog COM6M module is using a 38.4Mhz crystal

* Do not activate the BT enable pin during boot. this is done from the
  appliaction when BT is brought up.

Signed-off-by: Eyal Reizer <eyalr@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 56a61e0..8456b33 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -663,8 +663,7 @@ static int backlight_enable;
 
 struct wl12xx_platform_data am335xevm_wlan_data = {
 	.irq = OMAP_GPIO_IRQ(AM335XEVM_WLAN_IRQ_GPIO),
-	.board_ref_clock = WL12XX_REFCLOCK_26, /* 26 MHz */
-	.board_tcxo_clock = WL12XX_REFCLOCK_26, /* 26 MHz */
+	.board_ref_clock = WL12XX_REFCLOCK_38_XTAL, /* 38.4Mhz */
 };
 
 /* Module pin mux for wlan and bluetooth */
@@ -994,10 +993,8 @@ static void wl12xx_bluetooth_enable(void)
 	if (status < 0)
 		pr_err("Failed to request gpio for bt_enable");
 
-	pr_info("Enable bluetooth...\n");
+	pr_info("Configure Bluetooth Enable pin...\n");
 	gpio_direction_output(AM335XEVM_BT_ENABLE_GPIO, 0);
-	msleep(1);
-	gpio_set_value(AM335XEVM_BT_ENABLE_GPIO, 1);
 }
 
 static int wl12xx_set_power(struct device *dev, int slot, int on, int vdd)
-- 
1.7.11.2

